### DISCUSSION TITLE
Multilingual Termstore Search Results Not Working

### DISCUSSION THREAD

**Discussion by edwardkeay** [microsoft-search/pnp-modern-search#1867] [Q&A] [answered]
Hi,

I have an issue I can't seem to resolve regarding multilingual functionality and hoping someone can advise,

I am using modern search v4.5.4.0

I have a site with multilingual activated with English (base language) and French as an added language

On termstore items I have the language packs installed and have added French translations against terms, then created a site column that utlises the term set.

I have mapped the crawled property ows_taxid...  of the column to a RefinableString managed property, and selected the RefinableString property in the search results webpart

I have switched on **Enable localization** in the search results webpart, and set the **Language of search request** to be 'use interface language'

The scenarios are:

With **English** set in O365 and **English** set in the browser I search for the term "Link" (which is the English value in the multilingual term set) the correct results show.

With **French** set in O365 and **French** set in the browser I search for the term "Relier" (which is the French value in the multilingual term set) no results are found.

With **French** set in O365 and **French** set in the browser I search for the term "Link" (which is the English value in the multilingual term set) the correct results show.

The Filters are working fine and the correct language is displayed

With debug set in the layout section I can see all the tags, which includes the term "Relier" against the managed properties, although they do have the prefix Auto (unsure what that is??)

I have attached a screenshot showing results when **French** is set in O365 and **French** set in the browser, and I search for the term "Link" (English value), I have highlighted the properties with the localized values.

Why does the search text not find the word "Relier" when it clearly appears in the results??  
And is there any way to be able to search using words in a different language than the base langauge (English on my site) and have the correct results appear?

Thankyou in advance for any assistance with this

<img width="698" alt="results_search_screenshot" src="https://user-images.githubusercontent.com/67664032/155424007-35138f27-c3d6-4bff-aabb-8822f49a5983.png">

**Comment by wobba** [comment, marked as answer]
The web part translations is for displaying terms in the correct language only. See https://microsoft-search.github.io/pnp-modern-search/usage/search-results/data-sources/sharepoint-search/.

Results (and the search index) will only store the language of term matching the default language of the site where the item is. It's been like this since SP2013. Which means you can only get recall on the language of this term, not when querying in another language. There is no "synonym" support between languages - although a highly requested feature over the years.

**Reply by edwardkeay** [reply]
Hi,

Thankyou for response and information,

This is defintiely a feature that should beavailable, as it makes it properly multilingual

Thanks again

**Reply by wobba** [reply]
Would you expect terms in any termset to always be expanded to search for all language versions if a query term match? And what about alternate labels?

**Reply by edwardkeay** [reply]
Hi,

Yes I think search should be able to check all terms including different languages and synonyms

I have come up with a solution for my scenario.

I created a multi line text column, mapped it to a new managed property that is set to searchable.

Then using power automate, I make some http calls on each managed metadata column, which gets ALL terms in the associated termset including all languages and synonyms. 

I then append each found value to an array variable, which then I write to the multiline text field.

With the new column selected in the search results we part, it now returns the correct result if Relier is searched for.

**Reply by wobba** [reply]
Which works on a library by library case :)

**Reply by Jeroen00** [reply]
@wobba Should synonyms on a term within the same language work with search?

**Reply by wobba** [reply]
No.
